Add new enchantments from v. 1.3.4 update

nic0lette opened the pull request:
I updated the UI to use a property grid for items/enhancements rather than adding in a bunch of new NumericUpDowns. Basically it seemed like a good idea at the ... Okay, it sounded like an idea that I could use to get the update done fast without rewriting the UI to, for example, pull out the top High Scores section and put it somewhere else (like the "Bulk Edit" is)

But yes, hopefully the non-UI part isn't horrible -- this is one of my first projects in C# and trying to get it done in a few hours was... exciting XD

nic0lette committed:
Add new enchantments from v. 1.3.4 update

nic0lette committed:
Minor UI improvements

- Added auto-complete to the list of Pokémon because there are a lot of
  them and we really should know their names
- Fixed minor graphical glitch where the images for X & Y mega evolution
  stones were swapped

NinjaBoyLao commented:
Hey @nic0lette, does this tool now give the new items (basically the candy looking exp-up things)? Also, does it account for the new Zygarde Formes, as well as "Angry" Pikachu?

Commit:
Hopefully fixing 1.3.5 resources files access

nic0lette committed:
Merge pull request #1 from supercarotte/master

Master

Commit:
Merge remote-tracking branch 'refs/remotes/nic0lette/master'

Commit:
1.3.5 Update

Fixed mons.txt (again)
Updated database reading to 1.3.5
Fixed mega stages sprites
Updated Cheats to 1.3.5
"All Caught to Lvl 10" now sets level to current Max, based on number of
lollipops, rather than just 10.

nic0lette committed:
Merge pull request #2 from supercarotte/master

1.3.5 Update

nic0lette committed:
A few minor updates

- Bring back drag and drop
- Do better detection for save files instead of just the name
- Make sure we don't overflow levels when bulk editing

nic0lette committed:
Ensure Pokémon levels don't overflow

nic0lette committed:
Minor UI tweak

Commit:
Merge remote-tracking branch 'refs/remotes/nic0lette/master'

Commit:
Merge remote-tracking branch 'refs/remotes/nic0lette/master'

Commit:
OwnedBox Fix

Updated .bin files to 1.3.6 from extdata dump
Fixed the megastones sprites being inverted
Now sets the proper amount of lollipops everytime a pokemon's level is
edited
Updated Lv10 cheat into "LvMax" cheat with the same implementation

Commit:
WIP Mega-speedups

Added NUP fields for counting the number of speedups given to every
mega.
Plan to fix maximum values (found in pokemondata.bin) & add a patch
method

Commit:
Mega Speedups alpha version

Fetches maximum speedup values from pokemondata.bin
Added Bulk Edit code for speedups
Small UI adjustments

Known issue : Mega Mewtwo X speedup count circles between values 0 to 3
(instead of maximum = 5)

Commit:
Mega Speedups 1.0

Fixed M Mewtwo behaviour by ensuring Y data won't overwrite X data when
both share bits from the same byte.

Commit:
small commit

Added an icon because I felt like it

Commit:
v1.3 draft

Fixed PSSE not updating xp according to set level
Updated resource files to 1.3.7
Added a temporary icon because the default one is ugly

Commit:
Pull latest from supercarrotte

Commit:
1.3 draft fix

forgot to add PokemonLevel.bin, my bad

Commit:
Merge branch 'master' of github.com:supercarotte/PSSE

Commit:
Legitimacy Enhanced

Patched "Caught all available" code to also add event pokemons. Now
should only miss normal-type+opportunists+30BP (at the same time)
pokemons that would become available through event stages.
Disabled "All Caught" & "All Stones" codes by setting their visibility
to False. Set it back to Ture before building to enable them again (for
testing purposes only).

Commit:
Merge remote-tracking branch 'refs/remotes/nic0lette/master'

Commit:
Useless fancy stuff

Fixed stage images: now displays a different plate if the stage is a
mega pokemon (except for event stages, because they always have the
standard plate in game).

Added a team preview. You can click an icon to see the corresponding
pokemon in Owned box. No team editor because I think it's useless (team
cannot include uncatched pokemons).

Small UI adjustments to fit these changes.

Commit:
Stages update

Added rank visibles in Highscore boxes. Plain text for now because I
don't have the sprites.

Added "Complete all stages" & "S-Rank all completed stages" codes in
Bulk Edits window.

Small UI tweaks to fit those changes.

Commit:
Did someone say cheats ?

Added a bunch of bulk edit codes
"Escalation Battle step 999" sets the surrent escalation battle to step
999. Be careful, I don't know what happens if there are 2 escalation
battles at the same time.
"Delete StreetPass tags" sets SP encounters count to 0 & deletes all
your stored streetpass tags. This is because there was bug within
shuffle that'd make the game crash when entering the streetpass menu.
(buttons with a "/!\" symbol may behave strangely. Check message window
for more informations)

Added reset codes : respectively resets all Pokemon (uncatch, level,
stones, speedups & lollipops), stages (ranks, scores & completed status)
& resources (stock hearts, coins, jewels & all items/enhancements).

Fixed Owned Box :
-can own stone without owning the pokemon
-speedups now appear only if both caught & has stone
-uncatching a pokemon resets him to level 1, no lollipop & no speedup
-unchecking a megastone also reset the corresponding speedups counter
(to match legit in-game behaviours)

Commit:
Preparatory work

Cleanup from previous commit that might have been messed up.

Updated resource files to 1.3.8

Added the possibility to change ranks individually by clicking on the
rank label.

"All S rank" code now will update highscores instead of keeping them at
0. May update your highscore if it's inferior to (hitpoints + 2500),
a.k.a some fixed levels like #325 (known issue).

Commit:
v1.3 initial release

Fixed missing pokemon sprites

Fixed "All caught mega stones" to give only stones that have been
released.

Small code refactor

Commit:
v1.3.0-1

Cleaned a lot of code in Cheats().

Improved score editing function. "S-Rank Al Completed" code now sets
highscore to the lowest possible based on minimal S-Rank requirements
for that level (unless a higher score already exists).

Added Tooltips on Bulk Edits window. Keep your pointer for half a sec
over a button for a quick depiction of what it does and how to use it
for legitimacy purposes.

Bulk edits window now closes when Esc is pressed.

Circling between ranks for a stage is now done by clicking the picture
for that stage. I found it too hard to click directly on the label,
especially when it prints "-".

Fixed Rank editing for Event stages ("-" state wasn't reached when
circling).

Fixed Navigation with TAB Key, it should now make more sense (for both
main and bulk edits windows).

Commit:
Cleaning stuff

Fixed Deerling's incorrect forms sprites

Smalm optimizations

Commit:
Small work

Added a NUP field to display/edit lollipops given to a pokemon (only
appears if selected pokemon can be given one)

Click on a speedup/lollipop icon to switch between 0 and max values.

Fixed maximum speedup values (mistake from previous commit)

Fixed Items list not updating directly when a value was changed through
Bulk Edits

Commit:
Quick Fix

Fixed (?) Drag and Drop of savedata.bin files. Should also improve file
opening's behaviour.

Fixed Level & Lollipop NUP boxes. Now Lollipops automatically rise and
level automatically decreases if needed, that way you should always have
a legit level/lollipops pair.

Commit:
Great Refactor (and click events)

Refactored code, to be hopefully more C#/OOP-compliant.
Added a Database class that generates all resource files needed and is
accessible everywhere (no need to pass a lot of param to Cheats()
anymore).
Added a ToolFunctions class that provides generic functions to
search/edit savedata (formerly used in Cheats() only).

Added Click events
- On Pokemon picture : switches between Caught with all stones, max
level, lollipops & speedups, and Uncaught (with everything properly
reset).
-On Stages pictures : Left click to circle between ranks, Right click to
(un)complete stage.
-On "Main Stages" Label, switches on/off stage display override (see
below)

Now an uncompleted stage is displayed in black overlay (as in-game),
click "Main Stages" label to override this and see stage pictures
normally.

Also, now when a completed Main stage is uncompleted, all stages after
it are uncompleted too, and when an uncompleted stage is completed, all
stages before are completed (for legitimacy purposes, this is in testing
phase). Expert and Event stages aren't affected.

Small UI Tweaks

Commit:
Little Tweaks

Fixed Rank not updating properly.

Fixed Event stages not (un)completing properly.

Commit:
Keep refactoring

more code structure overhaul, I guess

Added a check/warning sprite to know if "Safe Mode" is active or not.
Safe Mode can now be switched on/off  by clicking on this sprite.

Unsafe Mode allows stages sprites to be fully colored even if they are
uncompleted, and prevents previous/next stages to be affected if you
(un)completed a stage.

Improved file opening.
Canceling the file selection or selecting an inapropriate file will keep
the previous file active.
If an appropriate file is being opened, an error message is displayed.

You can double click the FilePath TextBox at the top to eject the active
file from PSSE, reverting it to its default state.

Commit:
Another fix

Forgot to say that previous commit added a filter box in Open File
window (.bin files as default/all files)

Fixed Max speedups code in Bulk Edits to work only on pokémons for which
you own the stone.

code cleaning in cheats codes

Commit:
Keep refactoring

Mostly factorising methods

Commit:
1.3.10 Update

Update resource files to 1.3.10 version

Adds a Bulk Edit button for maxxing the talent's level of all caught
pokemons.
(Needs one more resource file added in next commit)

Commit:
1.3.10 fix : pokemonAbility.bin

Commit:
stupidest fix ever

Just pokemonAbility.bin not being properly creatd in resource folder...

Commit:
Keep refactoring (again)

As title says, it's still mostly refactoring.

Fixed "Complete all stages" Bulk code now properly ignores extra stages
that have 999 S ranks needed to unlock.

Commit:
Re-re-re-re-re-refactoring

Added methods to centralize ofsets/shifts definition.

Commit:
Tab navigating through Items

Also fixes Resources box overwriting values with old save values when a
new one is loaded.

Commit:
Added skill+ level viewer

in the form of a NUP field in Owned box, w/ a clickable icon (which
switches between level 1 or 5).
Also fixed incorrect skill+ experience values.
Also improved Tab navigation through the Items grid (the appriopriate
entry is now selected when entering the grid through Tab/Shift+Tab).

Commit:
Updated resource files to game's version 1.3.11

Also fixed skill level being properly set to 1/max when clicking the
pokemon's icon in Owned box.

Commit:
Fixes previous commit

Main.cs not properly updated due to a missclick...

Commit:
Quick fix for speedup handling

broke it with previous commits (didn't behave correctly w/ Mewtwo &
charizard because they had 2 stones)

Commit:
Update README.md

Commit:
Quick fix for Lollipop and Skill sprites reset

Just forgot to render them invisible on startup/reset.

Commit:
Added Form to choose values

to Fulfill smynos's request.

Most of the "Bulk Edits" buttons can now be Ctrl+clicked, which displays
a small form in which you can select the value to apply for that cheat
(instead of maximum).

Also removed the "blacked stage sprites" feature for now, I don't like
it anymore.

Also fixed the "SetEscalationStep" function : now it makes the passed
value the next step to beat instead of the last one beaten (giving 50 as
an arg would make you beat step 51 for example).

Commit:
Fixed handling of mega forms

db.HasMega now has 1 entry per pokemon rather than per specie, which
means that some alternate forms no longer display stones checkboxes
because their "normal" form has one.
They'll still display it if they have been given one though(shiny
gyarados, shiny rayquaza & winking glalie have one for example).

Commit:
Edit Pokathlon (dirty)

Included a Bulk Edit code to handle Survival mode (click/ctrl+click).
You need to have a save that has been paused during a survival mode
session for it to have an effect, as I have yet to find how to activate
that "pause" mode.

Added a "pokathlon.txt" file which contains the max/min values of legit
stages for each step (w/ shitty formatting, it's simply 2 lines (min
then max) per step).

Deleted "Reset Resources" button, now it's handled by Ctrl+click on
"(Max) Resources" button.

Now if you close one of the popups w/o pressing Enter or clicking "OK"
button, no edits will be done to your savefile (instead of applying the
default values).

I have yet to find how to enable/disable that "pause" mode for pokathlon
runs.

Commit:
Edit Pokathlon (better)

Added support for enabling the "paused" state that allow you to resume
your Pokathlon session.

Now "Edit Pokathlon" button sets step to 50, moves to 99 , opponent as
Mega Mewtwo Y and also activates the "paused" mode so that this session
can be resumed next time the game is launched.

Inverted "Random" button behaviour : now click it to get a random legit
opponent, or ctrl+click it to get a random opponent among the current
400 stages.

Commit:
Small fix

Inverted "Random" button behaviour again because I did it twice, it now
matches the previous commit's description.

Added some tooltips & changed the Pokathlon form's displayed name.

Commit:
Team editor beta

Added team editing to PSSE because in can be useful w/ Pokathlon edits.

Ctrl+clic a team sprite to copy active pokemon (from Owned Box) to that
slot.
Right-clic a team sprite (it goes a bit transparent) then right-clic
another to swap both slots.

Commit:
Cleanup

Also, previous commit fixed a mistake w/ Database.cs that prevented
"ctrl-click on active mon sprite" to correctly add speedups to some
pokemons.

Commit:
English fix

Fixed pokathlon edit confirmation displaying "11st", "12nd" or "13rd"

Commit:
Update bulit-in resource files to v1.3.13 of the game.

Commit:
Update README.md

Commit:
More options for StreetPass editing

"Delete Streetpass" button renamed to "Edit Streetpass (reset)".

Ctrl+clic it to set any number of streetpass encounters and decide if
you want to wipe your SP tags or not.

Commit:
Support for new Skill Swap item

Optimized ShuffleItems in case more items get added in the future.

Commit:
Add Skill Swapper description and fixes a deadly bug.

ItemsGrid's initialization defines an instance of ShuffleItems so it can
be seen in Design mode, but this instance doesn't update itself to
newest parameters.

Basically, that's something I don't know how to fix so it'll probably
stay that way.

Commit:
Added support for multiple skills

Resource files updated to 1.3.14 extdata dump. For some reason,
megastone.bin wasn't updated so there's an issue with shiny gyarados's
mega. I need to find where it has been added.

Added a Skill selector and updated a few functions so that they are able
to handle multiple skills.

Click on the Skill Booster icon sets all of your Pokémon's skills to
level 5 (if at least one isn't level 5) or 1.

Commit:
Fixed Shiny Gyarados editing

For each entry in megastone PSSE read the next ofset starting with Mega
Venusaur, which meant Smiling Gengar's max speedups were attributed to
Shiny Gyarados.
PSSE should now read the correct ofset.

Commit:
1.3.15 update + WIP Crystal hearts

Updated resource files to 1.3.15 version of the game.

Added a WIP cheat to give crystal hearts (7 renewable hearts + 700 coins
per day bonus). No idea what it breaks yet.

Removed "All stones and "All pokemons" hidden buttons. They are now
under the  single "Test stuff" hidden button, uncomment their respective
regions to activate again (still reserved to testing purposes).

Skill selector doesn't appear anymore if selected pokemon only has 1
skill.

Commit:
Mission cards support

Can now be edited using a bulk edit code : click to have every single
mission completed or ctrl+click for an editor to appear.

Still need to get the number of available mission cards & number of
missions for each card, from resource files.

Commit:
Improved Mission cards support

Added a new resource file : MissionCard.bin (which is arc/Configuration
Tables (B6680000)/12 in 1.3.15's files)

This allows the editor to display the correct amount of missions per
card and hide unreleased missions.

Commit:
Mission cards & WIP crystal hearts improvements

Crystal hearts WIP should be working, at least for this month.

Small Mission cards improvements.

While saving a file, its filename is properly displayed in the saving
window instead of it's full filepath.

Commit:
1.3.15 release

Nothing important here, just edited the tooltip about crystal hearts.

Commit:
Added verbose Skill output

Available skills for a Pokémonare now written in full text in a
selectable list.
Hover over that list for skill description.
This adds one resource file, "message_Pokedex_US.bin", from which the
strings are taken. It is found in Message_US folder.
English only for now, won't work with another region's strings.
Updated resource files to v1.3.17 of the game.

Commit:
1.3.20 release

Mostly updated resource files to game's 1.3.20 version
Fixed uncatching a Pokemon now resets its active skill to ites default
one.
Fixed Tab navigation behaviour.

Commit:
Fix for save canceling

Due to a syntax error, "updating" bool wasn't set to false which
prevented the save button to do anything after it had been canceled
once.
